exonum/src/proto/schema/exonum/runtime.proto

```diff
@@ -215,9 +215,25 @@ message InstanceMigration {
 message InstanceState {
   // Service instance specification.
   exonum.runtime.InstanceSpec spec = 1;
-  // Service instance activity status.
+  // Service instance activity status. 
+  //
+  // The `NONE` value means that it is initiated the addition of this instance to the blockchain
+  // and at the moment it is not yet completed. In this case, `pending_status` is `ACTIVE`.
+  //
+  // Note that status can be `NONE` only during the block execution.
   InstanceStatus status = 2;
-  // Pending status of instance if the value is not `NONE`.
+  // Pending status of instance.
+  //
+  // If this state is not `NONE`, then this instance is in the process of initiating addition,
```

Comment:
I'd write it something like:

"Pending state can be not `NONE` iff core is in process of changing service status, e.g. service initialization, resuming or migration. If this field was set to value other than `NONE`, it always will be reset to `NONE` in the next block."